### PR TITLE
Master imp extract

### PR DESCRIPTION
**Pull request by lmignon**
By default, all paths in a buildout, if not absolutes, are always
resolved as a child of ${buildout:diretory}. When extracting the sources
to a release directory it's therefore not possible to run the buildout
from outside the generated release directory. By prepending  paths of
the extracted sources with an overridable property, it's now possible
to execute buildout from outside of the release directory without
modification to the generated config file.

### CODE REVIEW

**Comment by sbidoul** [comment]
:+1: this works great and let us deploy the release in a subdirectory.

Having a separate directory for the release is important to clean-it up easily when deploying a new version (or deploy with rsync --delete).

**Comment by gracinet** [comment]
I'm neutral for now.

I still have more questions: would you also want this in the `a.r.openerp-1.9` branch ? If so, wouldn't it be wise to come up with a more name-agnostic name for the `odoo-src-dir` option ?

**Comment by lmignon** [comment]
Hi Georges,

What I try to cover with this change is to be able to run the buildout from outside the release directory in a way that let the release directory untouched after the run.
If I use the directory option to reference the release directory as buildout directory, the bin directory will be created inside the release directory.

> I still have more questions: would you also want this in the a.r.openerp-1.9 branch ? If so, wouldn't it be wise to come up with a more name-agnostic name for the odoo-src-dir option ?

Hard to find a name-agnostic name :smile:  'src-subdir',  'base-dir'?
About the branches policy, I m a little lost. What is the best place to contribute? At [acsone](http://acsone.eu) we use anybox.recipe.odoo.

**Comment by lmignon** [comment]
rebased

**Comment by petrus-v** [comment]
@lmignon 

I'll try to test this PR in the week.

For typo, what about `buildout-dir` ? also I wonder if it make sense to rename `to_src_path` method to `to_abs_src_path`?

About branch policy, I understand it's quite hard to follow!... basically we use anybox.recipe.odoo (this repo) for odoo >= v8 and anybox.recipe.openerp repo for odoo from 5.0 to 7.0.

I'm pretty sure, next release we will create a new branch for bug fixes for anybox.recipe.odoo.

I'm not sure yet what will be the policy for new feature that could be done in anybox.recipe.odoo and anybox.recipe.openerp! I guess developers will need to create a new PR against anybox.recipe.openerp if they want to see it on that repo as well.

We will discuss about that and try to make it clear.

Also I wonder what is the schedule for OCA migration. With this recipe under OCA umbrella I belive that would avoid to let PR open for a year... I understand it huge works to move responsibilities, @gurneyalex are you waiting information from Anybox? do you need help to do it? what is the state ?

**Comment by lmignon** [comment]
> For typo, what about buildout-dir ? 

What do you mean?

> also I wonder if it make sense to rename to_src_path method to to_abs_src_path?``

Ok to rename `to_src_path` in `to_abs_src_path`

**Comment by petrus-v** [comment]
@lmignon 

I mean using `buildout-dir` instead `odoo-src-dir`, I've red quickly the code, if I understand:

```
conf.set('buildout', 'odoo-src-dir', '${buildout:directory}')
```

`odoo-src-dir` is the absolute path to the buildout root directory, isn't it? so I suggest to rename to `buildout-dir`, does it make sense?

**Comment by lmignon** [comment]
@petrus-v 
`odoo-src-dir` is the property to override in the buildout.cfg in the  buildout-dir to specify the path to the release directory. 
for ex: in a /home/odoo-prod/instance/buildout.cfg
```
[buildout]
odoo-src-dir = /home/odoo-prod/instance/release
find-links = http://download.gna.org/pychart/
extends = release/release.cfg
parts = odoo

```

**Comment by petrus-v** [comment]
@lmignon 

Sorry for my bad understanding...

So can you write a bit of documentation about this setting?

I will do some experimentation later in the week, but why can't we use the `directory` buildout setting directly? is it because even `directory` is set in buildout.cfg we needs to get our cwd in buildout directory to run buildout?

Here a sample of what we are using on our server:
```
[buildout]
extends = /srv/odoo/test/current_buildout/release.cfg
directory = /srv/odoo/test/current_buildout

[openerp]
# don't try db_user = False, it does not work in some setuid situations
options.db_user = test
```
I don't remember if we run it from the buildout directory or from outside, I'll do this experimentation later.

**Comment by petrus-v** [comment]
Hooo maybe I get it, you extract all your code outside of the buildout directory so you have a tree like this:
```
├── odoo-prod/
│      └── instance/
│              ├── release/                # release directory
│              │      ├── parts/odoo       # odoo code
│              │      ├── addons           # addons path
│              │      └── ...
│              ├── buildout                # buildout directory
│              │      ├── bin/
│              │      ├── etc/
│              │      └── ....
│              └── buildout.cfg
```

is that correct?

**Comment by lmignon** [comment]
@petrus-v Yes it's

**Comment by gracinet** [comment]
I don't have time to reassess if I want to keep my promise to release 1.9.2b1 today.
@sbidoul, @lmignon : with what you've explained me of your workflow based on git-aggregator, do you still need it.

**Comment by sbidoul** [comment]
@gracinet no urgency for this one.
